include/NovelRT/Animation/SpriteAnimatorFrame.h

```diff
@@ -0,0 +1,40 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT License (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_ANIMATION_SPRITEANIMATORFRAME_H
+#define NOVELRT_ANIMATION_SPRITEANIMATORFRAME_H
+
+#ifndef NOVELRT_H
+#error Please do not include this directly. Use the centralised header (NovelRT.h) instead!
+#endif
+
+namespace NovelRT::Animation {
+  class SpriteAnimatorFrame {
+  public:
+    Utilities::Event<> FrameEnter;
+    Utilities::Event<> FrameExit;
+
+  private:
+    std::shared_ptr<Graphics::Texture> _texture;
+    float _duration;
```

Comment:
We should eventually change these to be some kind of `Timestamp` or `Timespan` like type, which tracks this in normalized ticks. Same I think for the `accumulatedDelta` in the `SpriteAnimator`.
Doing so helps avoid issues with floating-point over many frames.